test/fuzz/transformation_load_test.cpp

```diff
@@ -293,6 +293,176 @@ TEST(TransformationLoadTest, BasicTest) {
   ASSERT_TRUE(IsEqual(env, after_transformation, context.get()));
 }
 
+TEST(TransformationLoadTest, DemoTestCase) {
+  /* (will remove this after PR ready)
+  I want to edit 'TransformationLoadTest' to work for both OpLoad and
+  OpAtomicLoad
+  ---------------------------------------------------------------
+  OpLoad:
+  %res_type_id = opCode(OpLoad) %res_id %ptr_id MemOperand(literal)(optional)
+  ----------------------------------------------------------------
+  OpAtomicLoad:
+  %res_type_id = opCode(OpAtomicLoad) %res_id %ptr_id %scope_id
+  %semantic_id(mask - 32-bitIntegerScalar)
+
+  load through Pointer using give semantics.
+
+  - **ptr_id** for atomic load must be one of this:
+  Uniform when, StorageBufferm, PhysicalStorageBuffer,
+  Workgroup, CrossWorkgroup, Generic AtomicCounter, Image, Function.
+  Type of result type must be the same of type the pointer point to.
+
+  - %scope_id is the define the scope of execution, (32-bit integer scalar)
+  If labeled as a memory scope,
+  it specifies the distance of synchronization from the current invocation, what
+  this means?
+  Scopes:
+    -Everything executing on all the execution devices in the system.
+    CrossDevice = 0
```

Comment:
When we discussed scopes in our recent meeting I forgot to mention this: CrossDevice is for memory operations that should be visible right across the system - even when multiple GPUs are executing simultaneously.